### DISCUSSION TITLE
Path.toSVG() returns a string of svg for paths and such but without the <svg> </svg> tags wrapping it.

### DISCUSSION THREAD

**Discussion by teewuane** [fabricjs/fabric.js#9364] [Q&A] [answered]
When I call Path.toSVG() it returns a string of svg but without the actual `<svg></svg>` tag. When I call canvas.toSVG() it DOES include the `<svg>` and other tags. What is the reason for this? Is there a way to call toSVG() on a Path where it will return with the necessary xml tags?

**Comment by asturur** [comment, marked as answer]
the reason is that single objects toSVG are only meant to contribute to the larger SVG.
This is how it was designed.

A definetely easy way to get around this is to take the object you want to export, create a new static canvas (as big as the original one or as big as the object), clone the object there, export that canvas to svg.

Are you doing this in the browser or in node?

**Reply by teewuane** [reply]
Thank you, that makes sense! We are doing this in node.

**Reply by asturur** [reply]
Take care there are different solution too, that one could be expensive if canvas creation has to come into play in node.
In theory if no renders are necessary that is not expensive.

Other approaches are to hide every other object on the canvas and set a custom vieportTransform or viewbox, there is more than one if performance in node becomes an issue ( like canvas creation and trashing creates unnecessary memory pressure )
